packages/repeater/src/bus/DefaultRepeaterBus.ts

```diff
@@ -0,0 +1,199 @@
+import { RepeaterBus } from './RepeaterBus';
+import { RepeaterServer } from './RepeaterServer';
+import { RepeaterCommandHub } from './RepeaterCommandHub';
+import {
+  RepeaterErrorCodes,
+  RepeaterServerErrorEvent,
+  RepeaterServerEvents,
+  RepeaterServerReconnectionAttemptedEvent,
+  RepeaterServerReconnectionFailedEvent,
+  RepeaterServerRequestEvent,
+  RepeaterUpgradeAvailableEvent
+} from './RepeaterEventHub';
+import { Request } from '../request-runner/Request';
+import { Logger } from '@sectester/core';
+import chalk from 'chalk';
+
+export class DefaultRepeaterBus implements RepeaterBus {
+  private repeaterRunning: boolean = false;
+
+  constructor(
+    private readonly repeaterId: string,
+    private readonly logger: Logger,
+    private readonly repeaterServer: RepeaterServer,
+    private readonly commandHub: RepeaterCommandHub
+  ) {
+    this.repeaterServer.events.errorHandler = this.handleEventError;
+  }
+
+  public close() {
+    this.repeaterRunning = false;
+
+    this.repeaterServer.disconnect();
+
+    return Promise.resolve();
+  }
+
+  public async connect(): Promise<void> {
+    if (this.repeaterRunning) {
+      return;
+    }
+
+    this.repeaterRunning = true;
+
+    this.logger.log('Connecting the Repeater (%s)...', this.repeaterId);
+
+    this.subscribeToEvents();
+
+    await this.repeaterServer.connect(this.repeaterId);
+
+    this.logger.log('Deploying the Repeater (%s)...', this.repeaterId);
+
+    await this.deploy();
+
+    this.logger.log('The Repeater (%s) started', this.repeaterId);
+  }
+
+  private async deploy() {
```

Comment:
This method is unsafe; calling it may lead to memory leaks. Why do not register RepeaterServerEvents.CONNECTED's handler once in the `subscribeToEvents`?